syntax = "proto3";

option go_package = "pb/game";

// 服务配置，避免与 protobuf 的 Service 冲突
message ServiceConfig {
    string name = 1;  // 服务名称
    string url = 2;   // 服务的健康检查或访问URL
}

message BaseDeployConfig {
    repeated string etcd_hosts = 1;                   // Etcd 主机列表
    uint32 log_level = 2;                             // 日志级别
    repeated ServiceConfig services = 3;              // 服务列表
    repeated string service_discovery_prefixes = 4;   // etcd 服务前缀
    uint32 keep_alive_interval = 5;                   // 节点续约间隔（单位：秒）
    string deployservice_prefix = 6;                   // deployservice.rpc 前缀
    uint32 node_ttl_seconds = 7;                       // 节点租约TTL（单位：秒）
    uint32 health_check_interval = 8;                  // 新增：服务健康检查间隔（单位：秒）
}

// 游戏配置
message GameConfig {
    uint32 scene_node_type = 1;  // 场景节点类型，使用 uint32
    uint32 zone_id = 2;          // 区域ID，使用 uint32
	 // Zoneredis 配置
    message ZoneRedisConfig {
        string host = 3;           // Redis 服务器地址
        uint32 port = 4;           // Redis 服务器端口
        string password = 5;       // Redis 密码（可选）
        uint32 db = 6;             // 使用的 Redis 数据库编号
    }

    ZoneRedisConfig zone_redis = 7; // 添加一个 ZoneRedisConfig 配置
}
